walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -24,9 +24,31 @@
 public final class FindMeetingQuery {
   /* Finds all time ranges that satisfy the request given the already existing set of events. */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    // Find potential conflicts between scheduled events and request.
-    List<TimeRange> potentialConflicts = getPotentialConflicts(events, request);
+    // Find potential conflicts between scheduled events and request using only
+    // mandatory attendees.
+    Collection<String> mandatoryAttendees = request.getAttendees();
```

Comment:
Can you use ImmutableSet here?

I copied this from a Google document:
Why?
Immutable objects have many advantages, including:

Safe for use by untrusted libraries.
Thread-safe: can be used by many threads with no risk of race conditions.
Doesn't need to support mutation, and can make time and space savings with that assumption. All immutable collection implementations are more memory-efficient than their mutable siblings. (analysis)
Can be used as a constant, with the expectation that it will remain fixed.
Making immutable copies of objects is a good defensive programming technique. Guava provides simple, easy-to-use immutable versions of each standard Collection type, including Guava's own Collection variations.

The JDK provides Collections.unmodifiableXXX methods, but in our opinion, these can be

unwieldy and verbose; unpleasant to use everywhere you want to make defensive copies
unsafe: the returned collections are only truly immutable if nobody holds a reference to the original collection
inefficient: the data structures still have all the overhead of mutable collections, including concurrent modification checks, extra space in hash tables, etc.
When you don't expect to modify a collection, or expect a collection to remain constant, it's a good practice to defensively copy it into an immutable collection.

Reply:
Done - Thanks for the reference. I will have to copy it but from "it's a good practice to defensively copy it into an immutable collection" I think that is OK.